nbextensions/bulkImportCell/tabs/xsvGenerator.js

```diff
@@ -0,0 +1,221 @@
+            this.id = html.genId();
+
+            const state = this.model.getItem('state');
+            let errorMessage;
+            const hasErrors = Object.values(state.params).some((val) => val !== 'complete');
+            if (hasErrors) {
+                errorMessage =
+                    'Please note that there are errors in the input parameters for the selected data types, and that data upload will fail if the template is used unaltered.';
+            }
+
+            return div(
+                {
+                    class: `${cssBaseClass}__container`,
+                },
+                [
+                    div(
+                        {
+                            class: `${cssBaseClass}__container--errors`,
+                        },
+                        errorMessage ? p(errorMessage) : null
+                    ),
+                    // TODO: add form in here!
+                    p('Import types'),
+                    p('Output file type'),
+                    p('Destination directory'),
+                    div({
+                        id: this.id,
+                    }),
+                ]
+            );
+        }
+
+        /**
+         *  Coerce the form and file upload params into the appropriate format
+         *  for the staging service endpoint
+         *
+         * @param {object} formParams validated parameters from the form with keys
+         *      output_file_type: format to output the file in
+         *      output_directory: where to save the file
+         *      types: array of input types to include in the request
+         * @returns {object} output, suitable for sending to the staging service endpoint
+         */
+        generateRequest(formParams) {
+            const { output_file_type, output_directory, types } = formParams;
+
+            const params = this.model.getItem('params'),
+                appSpecs = this.model.getItem('app.specs'),
+                output = {
+                    output_directory,
+                    output_file_type,
+                    types: {},
+                };
+
+            types.forEach((fileType) => {
+                const appId = this.typesToFiles[fileType].appId;
+                output.types[fileType] = {
+                    order_and_display: appSpecs[appId].parameters.map((param) => {
+                        return [param.id, param.ui_name];
+                    }),
+                    data: params[fileType].filePaths.map((element) => {
+                        const datum = JSON.parse(JSON.stringify(element));
+                        Object.keys(params[fileType].params).forEach((param) => {
+                            datum[param] = params[fileType].params[param];
+                        });
+                        return datum;
+                    }),
+                };
+            });
+            return output;
+        }
+
+        /**
+         * Send the request to the staging service client
```

Comment:
started writing this method but it isn't finished / tested => commented out for now.

Reply:
ETA: since the method is so simple, I've uncommented it and added tests.